plugins/inputs/zfs/zfs_linux.go

```diff
@@ -60,30 +66,122 @@ func gatherPoolStats(pool poolInfo, acc telegraf.Accumulator) error {
 	keyCount := len(keys)
 
 	if keyCount != len(values) {
-		return fmt.Errorf("Key and value count don't match Keys:%v Values:%v", keys, values)
+		return fmt.Errorf("key and value count don't match Keys:%v Values:%v", keys, values)
 	}
 
 	tag := map[string]string{"pool": pool.name}
 	fields := make(map[string]interface{})
 	for i := 0; i < keyCount; i++ {
-		value, err := strconv.ParseInt(values[i], 10, 64)
+		value, err := strconv.ParseUint(values[i], 10, 64)
```

Comment:
Will the uint conversion cause issues for older versions of influx?

Reply:
good question! In my testing telegraf is emitting `field=###i`, regardless of the setting of `influx_uint_support` in `telegraf.conf`

Reply:
I did more testing and it does properly handle unsigned/signed based on `influx_uint_support`, but for some reason, not when running in `--test` mode... I'll leave that for another day